components/google_sheets/sources/common.js

```diff
@@ -42,7 +42,7 @@ module.exports = {
       this.db.set("subscription", { resourceId, expiration });
       this.db.set("channelID", channelID);
 
-      await this.takeSheetSnapshot();
+      this.db.set("isInitialized", false);
```

Comment:
On the first run, how do we know what changes have been made to a file if the content hasn't been initialized?

Reply:
On the first execution of `run()`, the flag will be false and it will kick in the initialization.
That's actually one of the fixes to avoid timing out during `activate()`. We basically put that burden on the `run()` method instead, which is OK if it takes a long time.